Basic-Car-Maintenance/Shared/Odometer/Views/AddOdometerReadingView.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  AddOdometerReadingView.swift
+//  Basic-Car-Maintenance
+//
+//  Created by Nate Schaffner on 10/15/23.
+//
+
+import SwiftUI
+
+struct AddOdometerReadingView: View {
+    
+    let vehicles: [Vehicle]
+    let addTapped: (OdometerReading) -> Void
+
+    @State private var date = Date()
+    @State private var selectedVehicle: Vehicle?
+    @State private var unitsAreMetric = false
+    @State private var distance = 0
+    @State private var switchUnitModalIsPresented = false
+    @Environment(\.dismiss) var dismiss
+    
+    var body: some View {
+        NavigationStack {
+            Form {
+                Section {
+                    HStack {
+                        TextField("Distance", value: $distance, format: .number)
+                        Button("Unit") {
+                            switchUnitModalIsPresented = true
+                        }
+                    }
+                    .sheet(isPresented: $switchUnitModalIsPresented) {
+                        Toggle(isOn: $unitsAreMetric) {
+                            Text(unitsAreMetric ? "Kilometers" : "Miles")
+                        }
+                        .presentationDetents([.height(100)])
```

Comment:
Instead of this, use a segmented picker
[See here](https://www.hackingwithswift.com/quick-start/swiftui/how-to-create-a-segmented-control-and-read-values-from-it)

Reply:
fixed in https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/175/commits/09829f2e93f16e3660276003b67ab25cae28f2b3